core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -56,7 +60,9 @@ class LogManager(val logDirs: Array[File],
                  scheduler: Scheduler,
                  val brokerState: BrokerState,
                  brokerTopicStats: BrokerTopicStats,
-                 time: Time) extends Logging {
+                 val zkUtils: ZkUtils,
```

Comment:
We could probably pass in a onLogDirFailure callback to the LogCleaner. An IOException could be triggered in different places. However, independent of where it's triggered, we always want to react with the same process, which includes (marking the partition/replica as offline, remove partition from replicaFetcher, notify the controller, etc). ReplicaManager seems to be the best place to consolidate that process.